sqlglot/expressions.py

```diff
@@ -5037,7 +5037,13 @@ def to_table(sql_path: t.Optional[str | Table], **kwargs) -> t.Optional[Table]:
     if not isinstance(sql_path, str):
         raise ValueError(f"Invalid type provided for a table: {type(sql_path)}")
 
-    catalog, db, table_name = (to_identifier(x) for x in split_num_words(sql_path, ".", 3))
+    catalog, db, table_name, *rest = (
```

Comment:
Thanks!

> Should we try to make clear that it's a project(id) that is the first part?

I think this terminology is BigQuery-specific, no? IIRC other dialects refer to this part in the name as the "catalog" and SQLGlot's just following that convention instead.

Reply:
Hi George. Indeed it's BQ specific. 

Ah I see. Not sure which providers support nested dots in table names, but I will take your word for it. In that case, looks good. Thanks again.

Reply:
Btw, found the relevant issue where you can see the discussion around the arbitrarily many Dots: https://github.com/tobymao/sqlglot/issues/489

Reply:
It's because the execution path is different. We've overrode the table part parsing logic for BigQuery because it has its own rules and since it now relies on `to_table`, there wasn't parity in the logic.